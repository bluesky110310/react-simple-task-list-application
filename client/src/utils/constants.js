export const STATUS_DATA = Object.freeze({
  TO_DO: "To Do",
  IN_PROGRESS: "In Progress",
  DONE: "Done",
});

export const STATUS_DATA_ALL = [
  STATUS_DATA.TO_DO,
  STATUS_DATA.IN_PROGRESS,
  STATUS_DATA.DONE,
];

export const DIALOG_DISPLAY_DATA = Object.freeze({
  HIDDEN_DIALOG: "Hidden",
  ADD_DIALOG: "Add",
  EDIT_DIALOG: "Edit",
});

export const BASE_URL = "http://localhost:5000";
