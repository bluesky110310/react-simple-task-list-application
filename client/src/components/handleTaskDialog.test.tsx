import { cleanup } from "@testing-library/react";
import { configure, shallow } from "enzyme";
import Adapter from "@cfaester/enzyme-adapter-react-18";
import { Button, DialogActions, DialogContent, TextField } from "@mui/material";
import HandleTaskDialog from "./handleTaskDialog";
import { DIALOG_DISPLAY_DATA, STATUS_DATA } from "../utils/constants";
import { Controller } from "react-hook-form";

afterEach(() => {
  cleanup();
});

configure({ adapter: new Adapter() });

const handleClose = jest.fn();
const manipulateTask = jest.fn();

const props1 = {
  displayType: DIALOG_DISPLAY_DATA.ADD_DIALOG,
  currentData: undefined,
  handleClose,
  manipulateTask,
};

const props2 = {
  displayType: DIALOG_DISPLAY_DATA.EDIT_DIALOG,
  currentData: {
    _id: "656c88a295a6bc188f6f1d4a",
    title: "Test",
    description: "This is the test task.",
    status: STATUS_DATA.IN_PROGRESS,
  },
  handleClose,
  manipulateTask,
};

const setup = (props: any) => {
  const wrapper = shallow(<HandleTaskDialog {...props} />);

  return { props, wrapper };
};

describe("HandleTaskDialog component", () => {
  test("it(ADD TASK) renders", () => {
    const { props, wrapper } = setup(props1);

    expect(wrapper).toMatchSnapshot();

    const textFields = wrapper.find(DialogContent).find(TextField);
    expect(textFields.at(0).props().id).toEqual("input-description");
    expect(textFields.at(0).props().value).toEqual("");

    const buttons = wrapper.find(DialogActions).find(Button);
    expect(buttons.at(0).text()).toEqual("Cancel");
    expect(buttons.at(1).text()).toEqual(props.displayType);
  });

  test("it(EDIT TASK) renders", () => {
    const { props, wrapper } = setup(props2);

    expect(wrapper).toMatchSnapshot();

    const textFields = wrapper.find(DialogContent).find(TextField);
    expect(textFields.at(0).props().id).toEqual("input-description");
    expect(textFields.at(0).props().value).toEqual(
      props.currentData.description
    );

    const buttons = wrapper.find(DialogActions).find(Button);
    expect(buttons.at(0).text()).toEqual("Cancel");
    expect(buttons.at(1).text()).toEqual(props.displayType);
  });
});
