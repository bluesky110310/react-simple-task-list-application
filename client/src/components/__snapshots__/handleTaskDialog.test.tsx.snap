// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HandleTaskDialog component it(ADD TASK) renders 1`] = `ShallowWrapper {}`;

exports[`HandleTaskDialog component it(EDIT TASK) renders 1`] = `ShallowWrapper {}`;
